### PR TITLE
consent

### PR DESCRIPTION
**Pull request by vlkerag**
## What are you changing?

- add @guardian/storage-and-cookies that does handle consent. Not in @guardian/libs to not have a circular dependency with cmp and libs

- using localStorage/sessionStorage becomes a lint error to encourage moving to @guardian/libs

## Why?

- To unify code usage and enable easier maintenance

### CODE REVIEW

File: libs/@guardian/eslint-config/index.js [hunk 1]
```diff
@@ -76,5 +76,17 @@ module.exports = {
 
 		// prevent circular dependencies
 		'import/no-cycle': ['error', { ignoreExternal: true }],
+
+		'no-restricted-syntax': [
+			'error',
+			{
+				selector: "CallExpression[callee.object.name='localStorage']",
+				message: 'Use @guardian/libs’s storage.local instead',
+			},
```
Comment (mxdvl) [inline libs/@guardian/eslint-config/index.js:85]: You also need to check for examples of `window.localStorage`, [here’s an ESLint playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXJlc3RyaWN0ZWQtc3ludGF4OiBbXCJlcnJvclwiLCBcIkNhbGxFeHByZXNzaW9uW2NhbGxlZS5vYmplY3Qub2JqZWN0Lm5hbWU9J3dpbmRvdyddW2NhbGxlZS5vYmplY3QucHJvcGVydHkubmFtZT0nbG9jYWxTdG9yYWdlJ11cIiwgXCJDYWxsRXhwcmVzc2lvbltjYWxsZWUub2JqZWN0Lm5hbWU9J2xvY2FsU3RvcmFnZSddXCJdICovXG5cblxud2luZG93LmxvY2FsU3RvcmFnZS5nZXQoXCJoZWxsb1wiKVxuXG5cbmxvY2FsU3RvcmFnZS5nZXQoXCJoZWxsb1wiKSIsIm9wdGlvbnMiOnsiZW52Ijp7fSwicnVsZXMiOnt9LCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e30sImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCJ9fX0=).

```suggestion
			{
				selector: "CallExpression[callee.object.name='localStorage']",
				message: 'Use @guardian/libs’s storage.local instead',
			},
			{
				selector:
					"CallExpression[callee.object.object.name='window'][callee.object.property.name='localStorage']",
				message: 'Use @guardian/libs’s storage.local instead',
			},
```

**Comment by vlkerag** [comment]
For concept discussion only at the moment, code is not functional as is

**Comment by sndrs** [comment]
@vlkerag i think we can close this now?

**Comment by sndrs** [comment]
(wrong button)

**Comment by sndrs** [comment]
closing this as a duplicate of #1128

**Pull request closed without merging**